src/main/webapp/3Dscript.js

```diff
@@ -1,31 +1,45 @@
-// import {FlyControls} from 'https://threejs.org/examples/jsm/controls/FlyControls.js';
 import {OrbitControls} from 'https://threejs.org/examples/jsm/controls/OrbitControls.js';
 import {GUI} from 'https://threejs.org/examples/jsm/libs/dat.gui.module.js';
-
+/**
+* A Point object with a lat and lng.
+* @typedef {Object<number, number, number>} Point
+* @property {number} lat The lat of the point.
+* @property {number} lng The lng of the point.
+* @property {number} alt The alt of the point.
+*/
 // These are global objects.
+/**
+ * @typedef {Array<number, number>}
+ * @property {Array<Point>} data All the stored run data for a given runId
+ * @property {string} color The color in a standard HTML acceptable format
+ */
+let runs;
+// Overall Threejs objects, may be helpful to condense into one Object.
 let scene;
 let camera;
 let renderer;
 let clock;
 let controls;
-let orientation;
-let trajectory;
-let pose;
-const poseTransform = {
+
+// Likely need to make a separate pose transform for each runId;
+const transforms = {};
```

Comment:
You can rename it runIdToTransform that can easily know it is a map from runId to its transform.

Good name can make your code have better readability.